### PR TITLE
Dev/tsweeney/b1.15.1

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
As the title says!

This was my first big cherry-pick and kit build on a release branch.  I'd problems with the cherry-pick, it didn't like one of the buildahimage/Containerfiles, so hopefully I didn't mess it up much.

### CODE REVIEW

**Comment by rhatdan** [comment]
Validate is not happy.
LGTM once test pass.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan I see everything in yellow here.  Where are you seeing an error?  But then again, I'm not sure why it's yellow 12+ hours later....

**Comment by TomSweeneyRedHat** [comment]
@rhatdan I cherry picked a number of Commits from your note, but three don't seem to be included.  Still looking, they are: 
chroot: do not use setgroups if it is blocked  - 2b1ad123a3ffdf4633a785fe800c2552da200d7a 
chroot, run: not fail on bind mounts from /sys  - c43ab204445141507aabbb2ea8c2e7e7585fc23e 
Mask over the /sys/fs/selinux in mask branch  - c00b434cd2fc3326911ad3d45d06af0fddf2fb39

**Comment by TomSweeneyRedHat** [comment]
OK, now they're all there.  I'm not sure why they didn't pull with the mass cherry-pick that I did to start.  Fingers crossed for the testing.

**Comment by TomSweeneyRedHat** [comment]
@nalind @rhatdan Gating is falling down with the below, but I don't think this commit is part of this PR, and if it is, I'm not sure what magic git commands I'd need to fix it?  I just checked my command history, I did a `git commit --amend -s` as part of the submit process.  Help!

```
[+0067s]  * f7ff4c11 "Merge pull request #2296 from TomSweeneyRedHat/bump-1.14.8" ... FAIL
[+0067s]   - FAIL - does not have a valid DCO
```

**Comment by nalind** [comment]
The validation test that runs as part of the gating job looks like it's picking an common ancestor commit from master as the baseline.  Going by `contrib/cirrus/test.sh`, it's using `$DEST_BRANCH` to direct the search for that ancestor.  Changing its value in `.cirrus.yml` from "master" to "release-1.15" looks to me like the right change to make to fix that.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,20 @@
...
+    Mask over the /sys/fs/selinux in mask branch
+    chroot: do not use setgroups if it is blocked
+    chroot, run: not fail on bind mounts from /sys
+    Allow "readonly" as alias to "ro" in mount options
+    Add VFS additional image store to container
+    Ignore OS X specific consistency mount option
+    vendor golang.org/x/text@v0.3.3
+    Cirrus: Fix missing htpasswd in registry img
+    Switch scripts to use containers.conf
+    Merge pull request #2471 from vrothberg/1.15-update-x/text
```
Comment (nalind) [inline CHANGELOG.md:15]: This is the merge commit log for line 12.  We don't need to mention it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
+    Make imagebuildah.BuildOptions.Architecture/OS optional
+    Merge pull request #2483 from nalind/release-1.15-arch
```
Comment (nalind) [inline CHANGELOG.md:17]: This is the merge commit log for the previous line.  We don't need to mention it.

**Comment by TomSweeneyRedHat** [comment]
@nalind, thx very much for the cirrus pointer.  I've touched that up and have removed the commit titles as you requested from the changelogs and buidlah.spec.

**Comment by TomSweeneyRedHat** [comment]
Unfortunately still the same DCO issue for that commit....

**Comment by nalind** [comment]
Based on the test log's `Linting & Validating from $GITVALIDATE_EPOCH to $GITVALIDATE_TIP` output, `contrib/cirrus/test.sh` is getting the right starting point from `git merge-base`.  The top-level `Makefile`'s `validate` target doesn't override that when the script invokes `make`.
`tests/validate/git-validation.sh` appears to override it to `c5546750573cdef524be7159fe837f23274e9718` if "$TRAVIS" isn't set, which looks like the rest of the problem.  That default should probably be getting set to the `release-1.15` branch's origin point (`3b1d6ebe12445dfe84cb5932634440a20b10fc03`) if "$CIRRUS_CI" isn't set.

**Comment by TomSweeneyRedHat** [comment]
thx again @nalind, tweaked and repushed.  I also added a comment in the git-validation.sh file explaining where the id there is getting pulled from.

**Comment by TomSweeneyRedHat** [comment]
@nalind, @cevich Looks like I'm finally past gating, now I'm hitting an artifact problem?
```
Uploading 1 artifacts for /var/tmp/go/src/github.com/containers/buildah/bin/*
Failed to upload artifact file /var/tmp/go/src/github.com/containers/buildah/bin/buildah: EOF
```
Thoughts?

**Comment by nalind** [comment]
LGTM

**Comment by nalind** [comment]
> @nalind, @cevich Looks like I'm finally past gating, now I'm hitting an artifact problem?
> 
> ```
> Uploading 1 artifacts for /var/tmp/go/src/github.com/containers/buildah/bin/*
> Failed to upload artifact file /var/tmp/go/src/github.com/containers/buildah/bin/buildah: EOF
> ```
> 
> Thoughts?

The branch's `.cirrus.yml` includes multiple `build_artifacts` keys that instruct Cirrus to grab the contents of `bin/*`, but the top-level `Makefile` doesn't put binaries there.  I'm not sure how a file that it failed to upload even got there.

**Comment by cevich** [comment]
> I'm not sure how a file that it failed to upload even got there.

One of the CI scripts creates that link for this purpose.  I think if there was some other primary-failure, failing to upload could be an expected secondary failure.  I would just ignore it unless it persists beyond the primary (build) failure.

**Comment by nalind** [comment]
Ah, the binaries are written to the `bin` directory on the `master` branch, but that wasn't the case yet when `release-1.15` was branched.  A dangling symlink might explain that error.

**Comment by TomSweeneyRedHat** [comment]
We're all green here now.  @cevich or @nalind, did one of you force a retest?  I'm not seeing evidence in the git logs above....

**Comment by nalind** [comment]
I pushed the retest button a couple of times, yes.

**Comment by TomSweeneyRedHat** [comment]
Ah, mystery solved, thanks @nalind.

**Comment by nalind** [comment]
> > I'm not sure how a file that it failed to upload even got there.
> 
> One of the CI scripts creates that link for this purpose. I think if there was some other primary-failure, failing to upload could be an expected secondary failure. I would just ignore it unless it persists beyond the primary (build) failure.

If this is a script rather than something Cirrus does because we set `binary_artifacts` in `.cirrus.yml`, I can't find it; can you point me to where this is done?  I got the same error in https://cirrus-ci.com/task/5209101888126976, which is for a PR that isn't on this branch.  Re-running the test made it succeed as well, but I'm no longer sure there isn't a problem we need to fix.

**Comment by vrothberg** [comment]
bors retry

**Comment by rhatdan** [comment]
bors  retry
@TomSweeneyRedHat once this merges please cut a release and get this in Podman. 2.0.4
@mheon FYI

**Comment by cevich** [comment]
> I can't find it; can you point me to where this is done?

Oh! Hmm, it was right around here https://github.com/containers/buildah/blob/master/contrib/cirrus/build.sh#L22

but it's gone now :confused: I swore I...oh, here's what I was thinking of:  https://github.com/containers/buildah/commit/90868620d#diff-43090a914b41249c6581b485cd9a4e06R22-R25

Does that help?

**Comment by nalind** [comment]
I guess we removed when we started putting the binaries there in #2380?  I don't know enough about the packaging system to know whether it was a requirement for making it work or if it was just a stylistic change that happened at the same time.

**Pull request merged**